File: eMartWEB/src/app/bean/GoodInfo.ts
export class GoodInfo {
    /**
     * 
     * @param status 0:normal,1:blocked,2.archived
     */
    constructor(
        public id?: string,
        public name?: string,
        public manufacturer?: string,
        public category?: string[],
        public detail?: string,
        public price?: number,
        public stock?: number,
        public count?: number,
        public owner?: string,
        public createdDate?: Date,
        public status: number = 0) { }

    clone(): GoodInfo {
        return new GoodInfo().init(this);
    }

    init(other): GoodInfo {
        for (let key in this) {
            this[key] = other[key];
        }
        return this;
    }

    equals(other) {
        for (let key in this) {
            if (this[key] !== other[key]) {
                return false;
            }
        }
        return true;
    }
}

File: emart-microservice/common-base/src/main/java/com/fsd/emart/common/bean/GoodInfo.java
package com.fsd.emart.common.bean;

import java.math.BigDecimal;
import java.sql.Timestamp;
import java.util.Arrays;
import java.util.List;

import com.fsd.emart.common.constants.Constants;
import com.fsd.emart.common.entity.ItemInfo;
import com.fsd.emart.common.util.StringUtil;

import lombok.Data;

@Data
public class GoodInfo {
    public String id;
    public String name;
    public String manufacturer;
    public List<String> category;
    public String detail;
    public BigDecimal price;
    public Integer stock;
    public String owner;
    public Timestamp createdDate;
    public Integer status;

    public static GoodInfo getInfoFromEntity(ItemInfo entity) {
        GoodInfo good = new GoodInfo();
        good.setId(entity.getItemId().toString());
        good.setName(entity.getName());
        good.setManufacturer(entity.getManufacturer());
        good.setCategory(Arrays.asList(StringUtil.splitString(entity.getCategory(), Constants.COMMA)));
        good.setDetail(entity.getDetail());
        good.setPrice(entity.getPrice());
        good.setStock(entity.getStock().intValue());
        good.setOwner(entity.getOwnerId());
        good.setCreatedDate(entity.getUpdateTime());
        good.setStatus(entity.getStatus());
        return good;
    }

}
